client/src/apis/backend/comments.js

```diff
@@ -0,0 +1,52 @@
+import axios from 'axios'
+import errHandler from './error'
+
+const service = axios.create({
+    baseURL: process.env.NODE_ENV === 'production' ?
+        '/api/comments' : `http://${window.location.hostname}:5000/api/comments`,
+    withCredentials: true,
+})
+
+export default {
+    searchComments() {
+        return service
+            .get('/search')
+            .then(res => res.data)
+            .catch(errHandler)
+    },
+
+    getComments() {
+        return service
+            .get('/')
+            .then(res => res.data)
+            .catch(errHandler)
+    },
+
+    addComment(body) {
+        return service
+            .post('/', body)
+            .then(res => res.data)
+            .catch(errHandler)
+    },
+
+    getComment(body) {
+        return service
+            .get(`/${body.id}`)
+            .then(res => res.data)
+            .catch(errHandler)
+    },
+
+    updateComment(body) {
+        return service
+            .patch(`/${body.id}`, body)
+            .then(res => res.data)
+            .catch(errHandler)
+    },
+
+    deleteComment(body) {
+        return service
+            .patch(`/${body.id}`, body)
+            .then(res => res.data)
+            .catch(errHandler)
+    },

```

Comment:
Shouldn't send body with the delete request.  Also should change .patch to .delete.
Same for other files too.